NDTensors/src/default_storage.jl

```diff
@@ -0,0 +1,39 @@
+## This is a fil which specifies the default storage type provided some set of parameters
+## The parameters are the element type and storage type
+function default_storage_type(
+  storetype::Type{<:TensorStorage}=Dense,
+  eltype::Type=Float64,
+  datatype::Type{<:AbstractArray}=Vector,
+  index=Tuple{Int},
+)
+  #TODO check if indices are block sparse or not
+  type = nothing
+  try
+    type = storetype{eltype,datatype{eltype}}
+  catch e
+    println("Cannot construct storetype", storetype)
+    throw(e)
+  end
+  return type
+end
```

Comment:
I would avoid this version of the function for now since the logic could become quite complicated making the element type, vector type, and storage type compatible with each other. For example, this would have to handle inputs like:
```julia
default_storage_type(Dense{Float32,Vector{Float32}}, ComplexF64, CuVector{ComplexF32})
```
which I think would benefit from a function like:
```julia
adapt_type(Float32, Vector{Float64}) -> Vector{Float32}
```
as we discussed the other day.

Reply:
Also maybe we should use `default_storagetype` to be consistent with the existing functions `storagetype` and `set_storagetype`.

Reply:
Yeah, I messaged the Julia slack and realized writing the function in this way made things much more difficult for myself. I have made some updates which should simplify the idea. I'm still going through your `Dense` comments